Bottom: 57c49bb54c28ffd41da0b641c52c54a95e397ab5
Top:    2712163046e0efb6e4e82cb93b94127aae088919
Author: Ian Kent <raven@themaw.net>
Date:   2013-05-02 14:37:41 +0800

autofs-5.0.7 - use MS_UNBINDABLE


---

diff --git a/daemon/direct.c b/daemon/direct.c
index 228a666..b1fd74a 100644
--- a/daemon/direct.c
+++ b/daemon/direct.c
@@ -420,7 +420,7 @@ int do_mount_autofs_direct(struct autofs_point *ap,
 
 	map_name = me->mc->map->argv[0];
 
-	ret = mount(map_name, me->key, "autofs", MS_MGC_VAL, mp->options);
+	ret = mount(map_name, me->key, "autofs", MS_UNBINDABLE|MS_MGC_VAL, mp->options);
 	if (ret) {
 		crit(ap->logopt, "failed to mount autofs path %s", me->key);
 		goto out_err;
@@ -768,7 +768,7 @@ int mount_autofs_offset(struct autofs_point *ap, struct mapent *me, const char *
 	if (!type || strcmp(ap->entry->maps->type, "hosts"))
 		map_name = me->mc->map->argv[0];
 
-	ret = mount(map_name, mountpoint, "autofs", MS_MGC_VAL, mp->options);
+	ret = mount(map_name, mountpoint, "autofs", MS_UNBINDABLE|MS_MGC_VAL, mp->options);
 	if (ret) {
 		crit(ap->logopt,
 		     "failed to mount offset trigger %s at %s",
diff --git a/daemon/indirect.c b/daemon/indirect.c
index f4476dc..ad5a82b 100644
--- a/daemon/indirect.c
+++ b/daemon/indirect.c
@@ -146,7 +146,7 @@ static int do_mount_autofs_indirect(struct autofs_point *ap, const char *root)
 	if (!type || strcmp(ap->entry->maps->type, "hosts"))
 		map_name = ap->entry->maps->argv[0];
 
-	ret = mount(map_name, root, "autofs", MS_MGC_VAL, options);
+	ret = mount(map_name, root, "autofs", MS_UNBINDABLE|MGC_VAL, options);
 	if (ret) {
 		crit(ap->logopt,
 		     "failed to mount autofs path %s at %s", ap->path, root);
